### PR TITLE
MI300 doc updates

### PR DESCRIPTION
**Pull request by mht-sharma**
Update doc to support MI300

Corresponding ORT Updates: https://github.com/huggingface/optimum/pull/1862

### CODE REVIEW

File: docker/onnx-runtime-amd-gpu/Dockerfile [hunk 1]
```diff
@@ -1,53 +1,67 @@
...
+RUN python3 -m pip install cmake onnx ninja transformers --no-cache-dir
+    
+RUN pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/rocm6.0 --no-cache-dir
 
 # Install ONNXRuntime from source
-RUN git clone --recursive  https://github.com/ROCmSoftwarePlatform/onnxruntime.git && cd onnxruntime && git checkout rocm5.7_internal_testing_eigen-3.4.zip_hash
+RUN git clone --single-branch --branch main --recursive https://github.com/Microsoft/onnxruntime onnxruntime
 
-RUN cd onnxruntime && ./build.sh --config Release --build_wheel --allow_running_as_root --update --build --parallel --cmake_extra_defines ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) --use_rocm --rocm_home=/opt/rocm
+RUN cd onnxruntime && ./build.sh --config Release --build_wheel --allow_running_as_root --update --build --parallel --cmake_extra_defines CMAKE_HIP_ARCHITECTURES=gfx90a,gfx942 ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) --use_rocm --rocm_home=/opt/rocm
```
Comment (fxmarty) [inline docker/onnx-runtime-amd-gpu/Dockerfile:62]: CMAKE_HIP_ARCHITECTURES=gfx90a,gfx942 is only for MI210/MI250/MI300 I think

**Review by fxmarty — APPROVED** [review]

File: docker/onnx-runtime-amd-gpu/Dockerfile (hunk 1 above)
Comment (mht-sharma) [reply docker/onnx-runtime-amd-gpu/Dockerfile:62]: yes, do we know what other archs are requested?

File: docs/source/amdgpu/overview.mdx [hunk 2]
```diff
@@ -45,15 +45,15 @@ with torch.device("cuda"):
...
 
-Hugging Face's [Text Generation Inference](https://huggingface.co/docs/text-generation-inference/index) library (TGI) is designed for low latency LLMs serving, and natively supports AMD Instinct MI210 and MI250 GPUs from its version 1.2 onwards. Please refer to the [Quick Tour section](https://huggingface.co/docs/text-generation-inference/quicktour) for more details.
+Hugging Face's [Text Generation Inference](https://huggingface.co/docs/text-generation-inference/index) library (TGI) is designed for low latency LLMs serving, and natively supports AMD Instinct MI210, MI250 and MI3O0 GPUs. Please refer to the [Quick Tour section](https://huggingface.co/docs/text-generation-inference/quicktour) for more details.
 
-Using TGI on ROCm with AMD Instinct MI210 or MI250 GPUs is as simple as using the docker image [`ghcr.io/huggingface/text-generation-inference:1.2-rocm`](https://huggingface.co/docs/text-generation-inference/quicktour).
+Using TGI on ROCm with AMD Instinct MI210 or MI250 or MI300 GPUs is as simple as using the docker image [`ghcr.io/huggingface/text-generation-inference:sha-XXXX`](https://huggingface.co/docs/text-generation-inference/quicktour).
```
Comment (mht-sharma) [inline docs/source/amdgpu/overview.mdx:54]: To update with the new docker image

File: docs/source/amdgpu/overview.mdx [hunk 3]
```diff
@@ -81,10 +81,12 @@ Check out more details about the support in [this guide](https://huggingface.co/
...
 
 ### AWQ quantization
 
-[AWQ](https://arxiv.org/abs/2306.00978) quantization, that is supported [in Transformers](https://huggingface.co/docs/transformers/main_classes/quantization#awq-integration) and [Text Generation Inference](https://huggingface.co/docs/text-generation-inference/basic_tutorials/preparing_model#quantization), is currently not available on ROCm GPUs.
+[AWQ](https://arxiv.org/abs/2306.00978) quantization, that is supported [in Transformers](https://huggingface.co/docs/transformers/main_classes/quantization#awq-integration) and [Text Generation Inference](https://huggingface.co/docs/text-generation-inference/basic_tutorials/preparing_model#quantization), is now supported on ROCm GPUs using Exllama kernels. With recent optimizations, the  AWQ model is converted to Exllama/GPTQ format model at load time. This allows ROCm and CUDA devices to benefit from the high quality of AWQ checkpoints and the speed of ExllamaV2 kernels combined.
```
Comment (guruprasad-mp) [inline docs/source/amdgpu/overview.mdx:88]: Can we change ROCm GPUs to AMD GPUs?

Comment (mht-sharma) [reply docs/source/amdgpu/overview.mdx]: Done!

**Pull request merged**